LoopKitUI/Views/Settings Editors/SuspendThresholdEditor.swift

```diff
@@ -12,59 +12,51 @@ import LoopKit
 
 
 public struct SuspendThresholdEditor: View {
-    var initialValue: HKQuantity?
-    var unit: HKUnit
-    var maxValue: HKQuantity?
-    var save: (_ suspendThreshold: HKQuantity) -> Void
-    let mode: SettingsPresentationMode
-    
+    @Environment(\.dismiss) var dismiss
+    @Environment(\.authenticate) var authenticate
+    @Environment(\.appName) private var appName
+
+    @ObservedObject var viewModel: TherapySettingsViewModel
+
     @State private var userDidTap: Bool = false
     @State var value: HKQuantity
     @State var isEditing = false
     @State var showingConfirmationAlert = false
-    @Environment(\.dismiss) var dismiss
-    @Environment(\.authenticate) var authenticate
-    @Environment(\.appName) private var appName
 
+    var initialValue: HKQuantity?
+    var maxValue: HKQuantity?
+    var save: (_ suspendThreshold: HKQuantity) -> Void
     let guardrail = Guardrail.suspendThreshold
 
     public init(
-        value: HKQuantity?,
-        unit: HKUnit,
-        maxValue: HKQuantity?,
-        onSave save: @escaping (_ suspendThreshold: HKQuantity) -> Void,
-        mode: SettingsPresentationMode = .settings
+        viewModel: TherapySettingsViewModel,
```

Comment:
The reason we have 2 `init`s here is historical: it was a solution for while the Prescription flow on-boarding was under development.  I don't think anyone went back and removed the first (this) one once it was no longer needed. Please get rid of this one and combine the logic in the other.  Thanks!

Reply:
@rickpasetto I see this will all therapy setting editors. Do you know if this is consistent? If so, I'll remove it every where. I have already marked them all as `fileprivate`.